### PR TITLE
Maximezmt/playbyartist

### PR DESCRIPTION
**Pull request by MaximeZmt**
Refactor the layout to choose the genre of music + add feature to query music by artist name in order to have more personalizable game 😀 ! 

photo_2022-05-17_22-46-35
photo_2022-05-17_22-46-38

### CODE REVIEW

File: app/src/main/res/layout/game_genre.xml [hunk 1]
```diff
@@ -5,97 +5,187 @@
...
+                    android:layout_height="130dp"
+                    android:layout_weight="1"
+                    android:background="@drawable/rounded_button"
+                    android:text="@string/imagine_dragons"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintHorizontal_bias="0.066"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    app:layout_constraintVertical_bias="0.687"></androidx.appcompat.widget.AppCompatButton>
```
Comment (jiabaow) [inline app/src/main/res/layout/game_genre.xml:44]: consider separating </androidx.appcompat.widget.AppCompatButton> in a new line

File: app/src/main/res/layout/game_genre.xml [hunk 2]
```diff
@@ -5,97 +5,187 @@
...
+                    android:layout_height="130dp"
+                    android:layout_weight="1"
+                    android:background="@drawable/rounded_button"
+                    android:text="@string/billie_eilish"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintHorizontal_bias="0.933"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    app:layout_constraintVertical_bias="0.687"></androidx.appcompat.widget.AppCompatButton>
```
Comment (jiabaow) [inline app/src/main/res/layout/game_genre.xml:63]: same here

File: app/src/main/res/layout/game_genre.xml [hunk 3]
```diff
@@ -5,97 +5,187 @@
...
+                    android:layout_height="130sp"
+                    android:layout_weight="1"
+                    android:background="@drawable/rounded_button"
+                    android:text="@string/r_o_c_k"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintHorizontal_bias="0.065"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    app:layout_constraintVertical_bias="0.403"></androidx.appcompat.widget.AppCompatButton>
```
Comment (jiabaow) [inline app/src/main/res/layout/game_genre.xml:87]: same

File: app/src/main/res/layout/game_genre.xml [hunk 4]
```diff
@@ -5,97 +5,187 @@
...
+                    android:layout_height="130dp"
+                    android:layout_weight="1"
+                    android:background="@drawable/rounded_button"
+                    android:text="@string/bts"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintHorizontal_bias="0.933"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    app:layout_constraintVertical_bias="0.402"></androidx.appcompat.widget.AppCompatButton>
```
Comment (jiabaow) [inline app/src/main/res/layout/game_genre.xml:106]: same

File: app/src/main/res/layout/game_genre.xml [hunk 5]
```diff
@@ -5,97 +5,187 @@
...
+                    android:layout_height="130dp"
+                    android:layout_weight="1"
+                    android:background="@drawable/rounded_button"
+                    android:text="@string/t_o_p_t_r_a_c_k_s"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintHorizontal_bias="0.074"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    app:layout_constraintVertical_bias="0.136"></androidx.appcompat.widget.AppCompatButton>
```
Comment (jiabaow) [inline app/src/main/res/layout/game_genre.xml:130]: same

File: app/src/main/res/layout/game_genre.xml [hunk 6]
```diff
@@ -5,97 +5,187 @@
...
+                    android:layout_height="130dp"
+                    android:layout_weight="1"
+                    android:background="@drawable/rounded_button"
+                    android:text="@string/k_p_o_p"
+                    app:layout_constraintBottom_toBottomOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintHorizontal_bias="0.937"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintTop_toTopOf="parent"
+                    app:layout_constraintVertical_bias="0.136"></androidx.appcompat.widget.AppCompatButton>
```
Comment (jiabaow) [inline app/src/main/res/layout/game_genre.xml:149]: also here

**Review by jiabaow — APPROVED** [review]
LGTM

File: app/src/main/java/ch/sdp/vibester/fragment/GameSetupFragment.kt [hunk 7]
```diff
@@ -221,6 +228,8 @@ class GameSetupFragment : Fragment(), View.OnClickListener, AdapterView.OnItemSe
             R.id.topTracksButton -> chooseGenre(method = LastfmMethod.BY_CHART.method, mode = R.string.top_tracks)
             R.id.billieEilishButton -> chooseGenre(method = LastfmMethod.BY_ARTIST.method, artist = "Billie Eilish", mode = R.string.billie_eilish)
 
+            R.id.validateSearch -> chooseGenre(method = LastfmMethod.BY_ARTIST.method, artist = searchArtistEditable.toString(), mode = R.string.byArtistSearch)
```
Comment (kamilababayeva) [inline app/src/main/java/ch/sdp/vibester/fragment/GameSetupFragment.kt:231]: why this one is not tested?

**Review by kamilababayeva — APPROVED** [review]
Thank you for a  nice change! I have left a small comment about testing a click button.

**Pull request merged**